### PR TITLE
fix: various style and UX fixes. TNL-10332.

### PR DESCRIPTION
**Pull request by kenclary**
fix: Correct summary of number of attempts; new separator between attempts summary and points summary; hint text for both attempts and points settings; fixed some widget paddings; advanced problems do not show type setting, and type setting menu does not show advanced problems. TNL-10332.

### CODE REVIEW

File: src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/ScoringCard.jsx [hunk 1]
```diff
@@ -14,11 +14,19 @@ export const ScoringCard = ({
...
+  };
+
   return (
     <SettingsOption
       title={intl.formatMessage(messages.scoringSettingsTitle)}
-      summary={intl.formatMessage(messages.scoringSummary,
-        { attempts: scoring.attempts.number, weight: scoring.weight })}
+      summary={getScoringSummary(scoring.attempts.number, scoring.attempts.unlimited, scoring.weight)}
     >
       <Form.Group>
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/ScoringCard.jsx:31]: Screen Shot 2023-01-20 at 11 07 40 AM
need to add this message here

**Review by KristinAoki — CHANGES_REQUESTED** [review]
When there are no hints the collapsed widget should read "None". Also `<HintRow>` should be left aligned with the header.

**Comment by KristinAoki** [comment]
Also can you change the title for your PR. Put all the things that you have changed in the PR description.

File: src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx [hunk 2]
```diff
@@ -29,7 +29,7 @@ export const HintsCard = ({
         />
       ))}
       <Button
-        className="my-3 ml-2"
+        className="ml-2"
```
[the remark this answers is not in the document]
Comment (kenclary) [reply src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx:32]: This seems reasonable, but is not covered by the jira issue. Is it accounted for somewhere else (I want to make sure I'm not doubling someone else's work)?

[the remark this answers is not in the document]
Comment (KristinAoki) [reply src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx:32]: I didn't see it in any of the tickets that jesper is working on

File: src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ export const HintsCard = ({
         />
       ))}
       <Button
-        className="my-3 ml-2"
+        className="ml-0 text-primary-500"
```
Comment (KristinAoki) [inline src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/settingsComponents/HintsCard.jsx:32]:
```suggestion
        className="pl-0 text-primary-500"
```

**Comment by KristinAoki** [comment]
> When there are no hints the collapsed widget should read "None". Also `<HintRow>` should be left aligned with the header.

To fix `<HintRow>` not being left aligned, I suggest changing them component to the following:
```
  <ActionRow className="mb-4">
      <Form.Control
        value={value}
        onChange={handleChange}
        floatingLabel={intl.formatMessage(messages.hintInputLabel)}
      />
    <ActionRow.Spacer />
    <IconButton
      src={DeleteOutline}
      iconAs={Icon}
      alt={intl.formatMessage(messages.settingsDeleteIconAltText)}
      onClick={handleDelete}
    />
  </ActionRow>
```

**Comment by kenclary** [comment]
Rewriting HintRow should probably be a separate PR, especially since it's not even in the original jira issue.

**Review by KristinAoki — APPROVED** [review]

**Review by KristinAoki — APPROVED** [review]

**Comment by jristau1984** [comment]
> Rewriting HintRow should probably be a separate PR, especially since it's not even in the original jira issue.

@kenclary just confirming - did this get a new ticket to be addressed?

**Comment by kenclary** [comment]
> > Rewriting HintRow should probably be a separate PR, especially since it's not even in the original jira issue.
> 
> @kenclary just confirming - did this get a new ticket to be addressed?

I recall that @KristinAoki integrated it into a later ticket. It did not get a *new* ticket.

**Pull request merged**